pkg/cmd/browse/browse.go

```diff
@@ -122,9 +129,13 @@ func NewCmdBrowse(f *cmdutil.Factory, runF func(*BrowseOptions) error) *cobra.Co
 	cmd.Flags().BoolVarP(&opts.WikiFlag, "wiki", "w", false, "Open repository wiki")
 	cmd.Flags().BoolVarP(&opts.SettingsFlag, "settings", "s", false, "Open repository settings")
 	cmd.Flags().BoolVarP(&opts.NoBrowserFlag, "no-browser", "n", false, "Print destination URL instead of opening the browser")
-	cmd.Flags().BoolVarP(&opts.CommitFlag, "commit", "c", false, "Open the last commit")
+	cmd.Flags().StringVarP(&opts.Commit, "commit", "c", "", "Select another commit by passing in the commit SHA, default is the last commit")
 	cmd.Flags().StringVarP(&opts.Branch, "branch", "b", "", "Select another branch by passing in the branch name")
 
+	// To preserve backwards compatibility when commit flag used to be a
+	// boolean flag.
+	cmd.Flags().Lookup("commit").NoOptDefVal = emptyCommitFlag
```

Comment:
I couldn't think of a better way of maintaining backwards compatibility while switching the commit flag from a boolean flag to a string flag.

This seemed like a suggested way of doing it by the pflag: https://pkg.go.dev/github.com/spf13/pflag#readme-setting-no-option-default-values-for-flags

However, this means that the commit flag now must use an equals sign to assign the flag value (i.e. `--commit=12a4`) which looks pretty inconsistent compared to the `--branch` flag, for example, which just expects a typical string argument after itself.

Please let me know if you think there's a way to improve this

Reply:
Yeah this is unfortunately the only way to do this. I think it is good for now and then in a future release, perhaps when we do another major release, we can remove the `NoOptDefVal` and break the backwards compatibility.